src/pages/userManagement/components/EditCollaborationDialog.jsx

```diff
@@ -0,0 +1,224 @@
+import React, { useEffect, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+
+import Alert from '../../../components/Alert';
+import Button from '../../../components/Button';
+import InputRow from '../../../components/fields/edit/InputRow';
+import SelectionEditor from '../../../components/fields/edit/SelectionEditor';
+import StandardDialog from '../../../components/StandardDialog';
+import Text from '../../../components/Text';
+import usePatchCollaboration from '../../../models/collaboration/usePatchCollaboration';
+import {
+  getSummaryState,
+  isEditApproved,
+  isViewApproved,
+  isViewRevoked,
+  states as collaborationStates,
+  summaryStates,
+} from '../../../models/collaboration/model';
+
+const coreStateChoices = [
+  {
+    value: 'view',
+    labelId: 'COLLABORATION_STATE_VIEW',
+  },
+  {
+    value: 'edit',
+    labelId: 'COLLABORATION_STATE_EDIT',
+  },
+];
+
+const fullStateChoices = [
+  ...coreStateChoices,
+  {
+    value: 'revoked',
+    labelId: 'COLLABORATION_STATE_REVOKED',
+  },
+];
+
+function FormattedUserName({ name }) {
+  return name || <FormattedMessage id="UNNAMED_USER" />;
+}
```

Comment:
can you use one of the userManagement utils in the file below? seems like this logic is getting repeated in a lot of different places in the app (not just this PR). i think we should lift this logic to the top of the app and apply it the same everywhere

Reply:
(though you dont need to do that to merge this PR)

Reply:
I made some changes. Let me know what you think:
1. There was already a FormattedReporter component from PR #444, so I updated AssetGroup and SightingEntityHeader to use it. (https://github.com/WildMeOrg/codex-frontend/pull/451/commits/2205d20605011b8c9d6d676e1ddec4d4c973bcb0)
2. I applied the `formatUserMessage` from this PR, which includes the email address in the result, to the collaboration management form. (https://github.com/WildMeOrg/codex-frontend/pull/451/commits/269a252bdbd35360a2632f1b0a719c5df34eb773)
    - I left `formatUserMessage` in the `pages/userManagement` directory because I think the name and email combination should only be used for user management right now. For some reason the `collaborationManagementForm` is not in that directory, but I think it should probably be in `userManagement/components`. I'll move it if you agree.
3. I removed this `FormattedUserName` component and instead crated a utility function in the general formatters file, also named `formatUserMessage` that just returns either the user's name or Unnamed User. (https://github.com/WildMeOrg/codex-frontend/pull/451/commits/94f708d01372f2cbfae95319e2368ba13277ebfd)

- I named both functions `formatUserMessage` because that is what they are doing. The user management one just needs more information. I could rename one, or expand the general formatter to handle including or ignoring the email address. I don't really want to put too much in one function right now because we seem to have a few similar, but slightly different needs for almost the same text.
- Within the user formatter functions and components, I used `fullName` because that is our convention for identifiers. However, because Houston sends `full_name` that means that in a lot, although not all, of the places the function is called, instead of just calling it with something like `formatUserMessage(user)`, we have to call it with `formatUserMessage({ fullName: user?.full_name })`.

Reply:
1. great!
2. great! i agree that `collaborationManagementForm` should be moved.
3. this is probably fine for now... eventually it could be good to build a single function that can accept a variety of different inputs but always tries to return a formatted user message. we can talk more about this later but i think it's a pattern we could use in a lot of places